### DISCUSSION TITLE
Add autochdir option

### DISCUSSION THREAD

**Discussion by smolck** [helix-editor/helix#10215] [Ideas]
#### Describe your feature request

In Neovim, there's an option `'autochdir'` that does the following:
```
'autochdir' 'acd'	boolean (default off)
			global
	When on, Vim will change the current working directory whenever you
	open a file, switch buffers, delete a buffer or open/close a window.
	It will change to the directory containing the file which was opened
	or selected.
	Note: When this option is on some plugins may not work.
```
I was wondering if something like this could be added to Helix?

(FWIW I actually managed [to get it somewhat working](https://github.com/smolck/helix/commit/180d8182f92cd19e3c2bdcf98f39d11ed9a4b576) but then realized what I had doesn't work for switching between splits, and probably several other edgecases I haven't considered, so figure it's probably best to make this issue about it.)

I will say that I've been using this option in Neovim for a while, but the more I think about it, this might not actually be a feature worth adding (and I might drop it from my neovim config actually). I'm honestly not sure, so interested to hear other's thoughts on this.

**Comment by the-mikedavis** [comment]
This could be interesting with something like the recently merged #1600. OTOH it seems like if I used it, I would end up having to think about which buffer I'm in before doing anything that takes the cwd into account which could be confusing.

**Comment by ghost** [comment]
Yes I would like this feature as well, Emacs works like this and you can have multiple projects open at the same time with different working directories

**Comment by noor-tg** [comment]
I think this is a good feature . could the code related to this config be merged to allow the intended behavior ?
the author of the issue can make pull request as in [autochdir](https://github.com/smolck/helix/commit/180d8182f92cd19e3c2bdcf98f39d11ed9a4b576) or even if he is not available I could make a pull request with the same code.

**Comment by uncomfyhalomacro** [comment]
> This could be interesting with something like the recently merged #1600. OTOH it seems like if I used it, I would end up having to think about which buffer I'm in before doing anything that takes the cwd into account which could be confusing.

Maybe if it is enabled, there is an option to show the directory of the buffer. So here are a possible config options for this feature:

Defaults:

```toml
[editor]
autochdir = false  # by default, it is false
```

If enabled:

```toml
[editor.autochdir]
show-directory = false
truncation-length = 2  # truncate the directory paths as desired
```

I want it to be disabled by default since some people do get confused when it is enabled by default

**Comment by uncomfyhalomacro** [comment]
There might be also an issue with the `file-picker` though. Maybe we could add an option to the `file-picker` too as well if we want it to respect the `autochdir` option or not from the `editor` section.

**Comment by Akselmo** [comment]
Is there any interested still in adding this? I have to traverse through a lot of different repos and projects when working on things, so it would be very helpful if the current working directory would change when i go to different project.

**Reply by Akselmo** [reply]
If so, I could perhaps try.. Never really written rust before though :sweat_smile:

**Reply by noor-tg** [reply]
I first straggled when try to make file relative to the current opened file. then I used the file lsp to open the file and save it. but it is not convenient like allow relative paths.
so if you can make it work do it.
for your info I tested this branch back. but it was not stable.

@Akselmo

**Comment by robgonnella** [comment]
If not this then could we consider an option to specify a "workspace" with paths to project sub-directories to include when loading configs at app launch? This would enable something like

```
workspace
├── .helix
├── deno_proj
│   └── .helix
└── node_proj
    └── .helix
```

where `workspace/.helix/config.toml` has something like

```toml
[workspace]
members = [ 'deno_proj', 'node_proj' ]
```

I might be wrong, but I think this would allow different language servers to be activated based on the project root of the current buffer and allows project root to be different from CWD.
